src/wallet/crypter.cpp

```diff
@@ -15,21 +15,40 @@
 #include <openssl/aes.h>
 #include <openssl/evp.h>
 
-bool CCrypter::SetKeyFromPassphrase(const SecureString& strKeyData, const std::vector<unsigned char>& chSalt, const unsigned int nRounds, const unsigned int nDerivationMethod)
+bool CCrypter::SetKeyFromPassphrase(
+        const SecureString& strKeyData,
+        const std::vector<unsigned char>& chSalt,
+        const unsigned int nRounds,
+        const unsigned int nDerivationMethod,
+        const std::vector<unsigned char> vchOtherDerivationParameters)
 {
     if (nRounds < 1 || chSalt.size() != WALLET_CRYPTO_SALT_SIZE)
         return false;
 
-    int i = 0;
-    if (nDerivationMethod == 0)
-        i = EVP_BytesToKey(EVP_aes_256_cbc(), EVP_sha512(), &chSalt[0],
-                          (unsigned char *)&strKeyData[0], strKeyData.size(), nRounds, chKey, chIV);
+    if (nDerivationMethod == 0) {
```

Comment:
... not that it really matters, because the encryption is authenticated, so trying to decrypt a wallet encrypted with a different algorithm will just fail.